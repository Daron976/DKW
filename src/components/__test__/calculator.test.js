import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../calculator';

describe('Calculator component', () => {
  test('header renders as exepcted', async () => {
    const snapCalculator = renderer.create(<Calculator />)
      .toJSON();
    expect(snapCalculator).toMatchSnapshot();
  });

  test('Calculator is present in the Dom', async () => {
    render(<Calculator />);

    const calcComponent = screen.getByTestId(/calculator/i);

    expect(calcComponent).toBeInTheDocument; //eslint-disable-line
  });

  describe('calculator UI', () => {
    test('clicked button is displayed', async () => {
      render(<Calculator />);

      const display = screen.getByTestId('display');
      const button9 = screen.getByText('9');
      fireEvent.click(button9);

      expect(display.innerHTML.trim()).toBe('9');
    });
    test('clicked button is displayed', async () => {
      render(<Calculator />);

      const display = screen.getByTestId('display');
      const button5 = screen.getByText('5');
      const button2 = screen.getByText('2');
      const button6 = screen.getByText('6');
      const button8 = screen.getByText('8');
      fireEvent.click(button5);
      fireEvent.click(button2);
      fireEvent.click(button6);
      fireEvent.click(button8);
      fireEvent.click(button2);

      expect(display.innerHTML.trim()).toBe('52682');
    });
    test('clicked button is displayed', async () => {
      render(<Calculator />);

      const display = screen.getByTestId('display');
      const button5 = screen.getByText('5');
      const button2 = screen.getByText('2');
      const addition = screen.getByText('+');
      const evaluation = screen.getByText('=');

      fireEvent.click(button5);
      fireEvent.click(addition);
      fireEvent.click(button2);
      fireEvent.click(evaluation);

      expect(display.innerHTML.trim()).toBe('7');
    });
  });
});
