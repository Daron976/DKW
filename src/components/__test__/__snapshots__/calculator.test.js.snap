// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component header renders as exepcted 1`] = `
<section
  className="flex"
  data-testid="calculator"
  id="calculator"
>
  <h1>
    Let's make some calculations
  </h1>
  <div
    className="calculatorWrapper flex"
  >
    <p
      className="screen flex"
      data-testid="display"
    >
       
       
    </p>
    <div
      className="btns flex"
    >
      <div
        className="numbers"
      >
        <div
          className="specialCharacters btn-grid"
        >
           
          <button
            className="symbol"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="symbol"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="symbol"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </div>
        <div
          className="firstRow btn-grid"
        >
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="7"
          >
            7
          </button>
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="8"
          >
            8
          </button>
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="9"
          >
            9
          </button>
        </div>
        <div
          className="secondRow btn-grid"
        >
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="4"
          >
            4
          </button>
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="5"
          >
            5
          </button>
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="6"
          >
            6
          </button>
        </div>
        <div
          className="thirdRow btn-grid"
        >
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="1"
          >
            1
          </button>
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="2"
          >
            2
          </button>
          <button
            className="number"
            onClick={[Function]}
            type="button"
            value="3"
          >
            3
          </button>
        </div>
        <div
          className="fourthRow"
        >
          <button
            className="number"
            id="double"
            onClick={[Function]}
            type="button"
            value="0"
          >
            0
          </button>
          <button
            className="symbol"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </div>
      </div>
      <div
        className="operators flex"
      >
        <button
          className="symbol"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="symbol"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="symbol"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="symbol"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="symbol"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</section>
`;
